tasks/main.yml

```diff
@@ -1,12 +1,18 @@
 ---
 
+- name: 'Create restic user and group'
+  include: 'create_user.yml'
+  when:
+    - restic_user != 'root'
+    - restic_group != 'root'
+
 - name: Ensure restic directories exist
   file:
     state: 'directory'
     path: '{{ item }}'
     mode: '0755'
-    owner: 'root'
-    group: 'root'
+    owner: '{{ restic_user }}'
```

Comment:
This is a security hole. Generally programs shouldn't be allowed to modify themselves and this adds such possibility. There is nothing wrong with a program with 755 permissions and `root` owner since mentioned permissions allow anyone to use this binary.

Reply:
The binaries themselves are owned by `root`

Reply:
Sorry, looks like I'm blind :smile:

Reply:
You can remove any files in directory if it's owned by you, so it's still insecure.

Reply:
@Nyoroon you are right, that's why I'm thinking of ditching a whole concept of installing binary to custom location and just installing it where it should be (at least according to [hier(7)](https://linux.die.net/man/7/hier)), in /usr/local/bin.